esp/esp/program/models/__init__.py

```diff
@@ -978,9 +978,9 @@ def getFloatingResources(self, timeslot=None, queryset=False):
             result = self.collapsed_dict(res_list)
             return [result[c] for c in result]
 
-    def getAvailableResources(self, timeslot):
+    def getAvailableResources(self, timeslot, queryset=False):
         #   Filters down the floating resources to those that are not taken.
-        return filter(lambda x: x.is_available(), self.getFloatingResources(timeslot))
+        return filter(lambda x: x.is_available(), self.getFloatingResources(timeslot=timeslot, queryset=queryset))
```

Comment:
The number of resources available seems to only pertain to a block. So if you have three laptops available, you can only check out 3 laptops in a block, but then if you go to block 2, even if those laptops weren't returned, you can still check out 3 more laptops, at least from what I can tell. The resource numbers available should account for the resources checked out during previous blocks that havent been returned yet. (Important because teachers dont return from block 1 until after block 2 teachers have already checked in)

Reply:
Hmm...this will not be easy due to how resources work. Resources (and resource assignments) are timeslot-specific, so they don't really know about resources of other timeslots. I guess what I'm saying is it will be pretty hacky to make this work (and in my mind, if all of a resource is already checked out and not returned in real life, an admin would be smart enough to not check it out to another teacher).

Reply:
I guess that's a fine assumption for the time being then!

Reply:
OK, I'm happy to return to this at some point, because I do think it would be a nice thing to include, but I don't think the investment will be worth the return at this point.